### ISSUE TITLE
avrdude.conf can't be found on fedora distribution

### ISSUE THREAD

**Issue by slevirus** [amperka/ino#51]
avrdude package not have avrdude.conf in /usr/share/arduino but in /etc/avrdude/avrdude.conf. Just hack path finding for add repertory for all fedora distribution.
Nice work and thank you

**Comment by slevirus** [comment]
```
diff --git a/ino/commands/upload.py b/ino/commands/upload.py
index e081219..ca36772 100644
--- a/ino/commands/upload.py
+++ b/ino/commands/upload.py
@@ -39,7 +39,10 @@ class Upload(Command):
         self.e.find_tool('stty', ['stty'])
         if platform.system() == 'Linux':
             self.e.find_arduino_tool('avrdude', ['hardware', 'tools'])
-            self.e.find_arduino_file('avrdude.conf', ['hardware', 'tools'])
+            if platform.dist()[0] == 'fedora':
+                self.e.find_arduino_file('avrdude.conf', [])
+            else:
+                self.e.find_arduino_file('avrdude.conf', ['hardware', 'tools'])
         else:
             self.e.find_arduino_tool('avrdude', ['hardware', 'tools', 'avr', 'bin'])
             self.e.find_arduino_file('avrdude.conf', ['hardware', 'tools', 'avr', 'etc'])
diff --git a/ino/environment.py b/ino/environment.py
index 92da645..482df67 100644
--- a/ino/environment.py
+++ b/ino/environment.py
@@ -63,8 +63,10 @@ class Environment(dict):
         '/usr/local/share/arduino',
         '/usr/share/arduino',
     ]
-
-    if platform.system() == 'Darwin':
+    if platform.system() == 'linux':
+        if platform.dist()[0] == 'fedora':
+            arduino_dist_dir_guesses.insert(0, '/etc/avrdude')
+    elif platform.system() == 'Darwin':
         arduino_dist_dir_guesses.insert(0, '/Applications/Arduino.app/Contents/Resources/Java')

     default_board_model = 'uno'
```

**Comment by nkrkv** [comment]
Thanks for the report. There should be a way to get to know where `avrdude.conf` actually is. Since Arduino IDE itself knows it somehow. Will investigate the issue.

**Comment by slevirus** [comment]
I think it's hardcoded. It's a specific distribution issues.

```
cd /usr/share/arduino/
grep -nr 'avrdude.conf' .
./hardware/arduino/bootloaders/optiboot/Makefile:55:AVRDUDE_CONF = -C$(TOOLROOT)/avr/etc/avrdude.conf
./hardware/arduino/bootloaders/optiboot/Makefile:77:AVRDUDE_CONF = -C$(TOOLROOT)/avr/etc/avrdude.conf
./hardware/arduino/bootloaders/caterina/Makefile:438:AVRDUDE = /Applications/avrdude -C /Applications/avrdude.conf -B 1
```

If it's can help you
bye 
:-)

**Comment by brianredbeard** [comment]
Any traction on this?  Pretty soon we'll have to start maintaining patches in the RPM.

**Comment by nkrkv** [comment]
Oh, well, sorry for the long delay. Will find a bit of spare time for this in few days. Thank you for bothering.

**Comment by nkrkv** [comment]
Fixed in 0.3.6
